lib/protocol/ConnectionManager.js

```diff
@@ -13,52 +13,49 @@
 // language governing permissions and limitations under the License.
 'use strict';
 
-var State = require('./ConnectionManagerState');
 var util = require('../util');
 var debug = util.debuglog('hdb_conn');
 
 module.exports = ConnectionManager;
 
 function ConnectionManager(settings) {
-  this.state = new State(settings);
+  this._settings = settings;
 }
 
-ConnectionManager.prototype.updateState = function updateState(newOptions) {
-  this.state.update(newOptions);
-};
-
-ConnectionManager.prototype.getConnectOptions = function getConnectOptions() {
-  return this.state.options.connect;
+ConnectionManager.prototype.updateConnectOptions = function updateConnectOptions(settings) {
+  return util.extend(this._settings, settings);
```

Comment:
`util.extend(this._settings, settings);` changes `this._settings`.
It would be better if this class has 2 separate members:
- `initialSettings`
- `currentSettings`

That's because hdb driver supports the following:

```js
var options1 = {};
var options2 = {};
var options3 = {};

var client = hdb.createClient(options1);
client.connect(options2, function() {
  // ...
  client.close();
  client.connect(options3, function() {
    // ...
    client.close();
  });
});
```

We better keep the initial options (`options1`) without changing them so we can later use them if the client wants to connect again. And on each connect we will reassign the `currentSettings`.